test/org/sagebionetworks/bridge/play/controllers/SurveyControllerTest.java

```diff
@@ -951,18 +652,15 @@ private void assertCacheIsCleared(ExecuteSurvey executeSurvey) throws Exception
 
         // Now mock the service because the *next* call (publish/delete/etc) will require it. The 
         // calls under test do not reference the cache, they clear it.
-        when(service.getSurvey(any(), eq(true))).thenReturn(survey);
         when(service.publishSurvey(any(), any(), anyBoolean())).thenReturn(survey);
-        when(service.versionSurvey(any())).thenReturn(survey);
-        when(service.updateSurvey(any())).thenReturn(survey);
+        when(service.versionSurvey(eq(TestConstants.TEST_STUDY), any())).thenReturn(survey);
+        when(service.updateSurvey(eq(TestConstants.TEST_STUDY), any())).thenReturn(survey);
         
         // execute the test method, this should delete the cache
         executeSurvey.execute(SURVEY_GUID, CREATED_ON.toString());
-        verify(service).getSurvey(any(), eq(true));
         
         // This call now hits the service, not the cache, for a total of two calls to the service
         controller.getSurvey(SURVEY_GUID, CREATED_ON.toString());
-        verify(service, times(2)).getSurvey(any(), eq(true));
```

Comment:
Should we still verify caching logic in our unit tests?

Reply:
Added this back corrected (we now call surveyService once).